compiler/prelude/jsmapping.go

```diff
@@ -132,7 +135,11 @@ var $externalize = function(v, t) {
       if (f.pkg !== "") { /* not exported */
         continue;
       }
-      o[f.name] = $externalize(v[f.prop], f.typ);
+      var name = f.name;
+      if (name === "") {
+        name = f.prop;
```

Comment:
We should probably not do this, because `f.prop` is an internal name and may be anything, e.g. it might be minified. What about doing the same as `encoding/json`: `Anonymous struct fields are usually marshaled as if their inner exported fields were fields in the outer struct, subject to the usual Go visibility rules`